### ISSUE TITLE
Edtext inside sameword gives error

### ISSUE THREAD

**Issue by stenskjaer** [maieul/ledmac#767]
Issue can be opened in English or in French (I prefer in French). Please fill this form.
Please open one issue by problem, and not a global issue for multiple problems. 

# Version of reledmac / reledpar (if that is a devellopement version, please precise the branch)
2.24.2

# If this is a newly introduced problem, please indicate the last working version of the package(s)

# Does the problem happen only when combining with some packages?
No.

# Does the problem happens both in parallel typesetting and normal typesetting?
Yes.

# Description of the problem
When a `\sameword{}` entry contains an `\edtext{}{}` element, the compilation gives the error "! Missing \endcsname inserted." on the line of the nested encoding. The problem is also described in https://github.com/stenskjaer/samewords/issues/33 with a bit more context and more complex examples.

# Minimal Working Example
Please provide a [MWE](http://minimalbeispiel.de/mini-en.html). 
In  case where the problem happens both in parallel typesetting and normal typesetting, please make your MWE in normal typesetting (because your MWE must be _Minimal_). 

``` latex
\documentclass{article}

\usepackage[series={A},nofamiliar,noeledsec,noledgroup,draft]{reledmac}

\begin{document}

\beginnumbering
\pstart
\sameword{two \sameword{dollars}}
\sameword{two \edtext{\sameword[2]{dollars}}{\Afootnote{cents}}}{\Afootnote{del.}}
\pend
\endnumbering

\end{document}
```

**Comment by maieul** [comment]
I should be solvable, I need juste some time.

**Comment by maieul** [comment]
Ok
1. First of all, you had an error in your MWE: you have a \Afootnote outside of an edtext. Cf 69e0d0532d59591dd6e24d288e783683793c7963 and dcc75c1a
2. Then, I have modified reledmac in branch issue767:
    - Fix bug with optional argument of \sameword inside a \saweword
    - Allow \edtext inside \sameword
Please try it and let me know.

Also, I have tried with nestedt sameword/edtext (dcc75c1a).  But with the default way of printing sameword, that was unclear, because we get two 1 and two 2,  the first refering to "two dollars" and in the second refering to "dollars".  But I found a way to make parenthesis around a the samword, in the foonote, when concerning two words or mord. cf fedd07f1. 

I don't think I will integrate it direclty in reledmac, as I thinkg it is not a good idea to refer to two words as a single sameword.

**Comment by stenskjaer** [comment]
Thanks for looking at this!

Apologize for the mistakes in the MWE. It was the result of a simplification from this, more complex, example:

``` latex
\documentclass{article}

\usepackage[series={A},nofamiliar,noeledsec,noledgroup,draft]{reledmac}

\begin{document}

\beginnumbering
\pstart
\sameword{\sameword{two} \sameword{dollars}} and
\edtext{\sameword[1]{two}}{\Afootnote{four}} cents and
\edtext{\sameword[1]{\sameword{two} \edtext{\sameword[2]{dollars}}{\Afootnote{cents}}}}{\Afootnote{del.}}
and more \sameword{dollars}.
\pend
\endnumbering

\end{document}
```

With the reledmac from branch "issue767" this compiles. 

But the apparatus looks like this:
```
1 two² ] four
1 two³ dollars²² ] del. 
1 dollars² ] cents
```
This is of course because it numbers both the single and multi-word `\sameword{}`. 
What I would like to see is this:
```
1 two² ] four
1 two dollars² ] del. 
1 dollars² ] cents
```
Where the "two dollars²" is because the whole expression "two dollars" occurs two times in the line, and we refer to the second of these. So it means something like "(two dollars)²" and not "(two) + (dollars)²".

I realize that you do not find such reference to more than one word clear, and that you would personally prefer something like "two³ dollars²" (we also discussed that before in https://github.com/maieul/ledmac/issues/649). This can be done, and works fine.
I do however not find "two dollars²" more unclear, as I read "(two dollars)²" and not "(two) + (dollars)²". But I do find it much more pretty. 
And I believe that both approaches match the intuition of different readers and editors. 

In the above case I do however not see any correct or good way to get what I would like ("two dollars²") and still get correct numbering in the other references.

*IDEA*
How about using the optional level numbering more to determine which `\sameword{}` applies to the apparatus note? Take this snippet:
```latex
\edtext{\sameword[1]{\sameword{two} \edtext{\sameword[2]{dollars}}{\Afootnote{cents}}}}{\Afootnote{del.}}
```
Here the content of `\sameword[1]{...}` is relevant for the last app note. So in printing the numbers it should ignore printing of numbers on all `\sameword{}`s that do not match that (i.e. have `\sameword[1]`). This would mean that the sameword-annotation of `\sameword{two}` and `\sameword[2]{dollars}` should not result in a printing of a number, as the level argument does not match.

This would of course require an editor to use the optional argument to indicate the level to work. But I honestly also think that it is a good idea to always use that anyway, and if the editor uses my script, that is how it will be done. 

This is just my view of it, of course. What do you think about cases like this, @floriandk??

**Comment by maieul** [comment]
Indeed, optional argument what thoug, at first step, for dealing with the problem of `\lemmacommand`. Your solution could be a good idea, but I would prefer to kept it as an option, especially because I am not sure that I could not create ambigouus case. Also because of ascendant compatibility.  It should be easy to install (just modifying the test l. 1954 of current reledmac.sty). But in any case, that is an other topic, and I would ask you to open an other issue.

**Comment by maieul** [comment]
ps : an other solution, as explained in my first answer, is https://github.com/maieul/ledmac/commit/fedd07f1c65f68dec641b55d8c48261c214489f1

**Comment by stenskjaer** [comment]
All right, fair enough. I will create a new issue about the multi-word annotation. You can close this one if you want, as it solves the original problem.

Thanks!

**Comment by maieul** [comment]
I have published a release on CTAN, just now.
